src/stream/src/executor/over_window/frame_finder.rs

```diff
@@ -890,4 +1160,424 @@ mod tests {
+
+        use super::*;
+
+        fn create_range_frame<T>(
+            order_data_type: DataType,
+            order_type: OrderType,
+            start: FrameBound<T>,
+            end: FrameBound<T>,
+        ) -> RangeFrameBounds
+        where
+            T: Into<ScalarImpl>,
+        {
+            let offset_data_type = match &order_data_type {
+                t @ data_types::range_frame_numeric!() => t.clone(),
+                data_types::range_frame_datetime!() => DataType::Interval,
+                _ => unreachable!(),
+            };
+
+            let map_fn = |x: T| {
+                RangeFrameOffset::new_for_test(x.into(), &order_data_type, &offset_data_type)
+            };
+            let start = start.map(map_fn);
+            let end = end.map(map_fn);
+
+            RangeFrameBounds {
+                order_data_type,
+                order_type,
+                offset_data_type,
+                start,
+                end,
+            }
+        }
+
+        #[test]
+        fn test_calc_logical_for_int64_asc() {
+            let order_data_type = DataType::Int64;
+            let order_type = OrderType::ascending();
+
+            let range_frames = [
+                &create_range_frame(
+                    order_data_type.clone(),
+                    order_type,
+                    Preceding(3i64),
+                    Preceding(2i64),
+                ),
+                &create_range_frame(
+                    order_data_type.clone(),
+                    order_type,
+                    Preceding(1i64),
+                    Following(2i64),
+                ),
+            ];
+
+            let ord_key_1 = StateKey {
+                order_key: memcmp_encoding::encode_value(&Some(ScalarImpl::Int64(1)), order_type)
+                    .unwrap(),
+                pk: OwnedRow::empty().into(),
+            };
+            let ord_key_2 = StateKey {
+                order_key: memcmp_encoding::encode_value(&Some(ScalarImpl::Int64(3)), order_type)
+                    .unwrap(),
+                pk: OwnedRow::empty().into(),
+            };
+
+            let (logical_first_curr, logical_last_curr) =
+                calc_logical_curr_for_range_frames(&range_frames, &ord_key_1, &ord_key_2).unwrap();
+            assert_eq!(
+                logical_first_curr.as_normal_expect(),
+                &Some(ScalarImpl::Int64(-1))
+            );
+            assert_eq!(
+                logical_last_curr.as_normal_expect(),
+                &Some(ScalarImpl::Int64(6))
+            );
+
+            let (first_start, last_end) =
+                calc_logical_boundary_for_range_frames(&range_frames, &ord_key_1, &ord_key_2)
+                    .unwrap();
+            assert_eq!(first_start.as_normal_expect(), &Some(ScalarImpl::Int64(-2)));
+            assert_eq!(last_end.as_normal_expect(), &Some(ScalarImpl::Int64(5)));
+        }
+
+        #[test]
+        fn test_calc_logical_for_timestamp_desc_nulls_first() {
```

Comment:
The timestamp format seemed to be supported actually, but why it's disabled in the planner test?

Reply:
`timestamp` and `timestamptz` are supported. While for `timestamptz`, we only support offset to be interval without `month` and `day` fields, because that requires timezone info to do calculation and that is not currently supported by our `Add(Timestamptz, Interval)` expression.